canonical/prometheus-pushgateway-k8s-operator
# Loki integration

## Issue


This PR add Loki integration to Pushgateway


## Testing Instructions


1. Run itest

2. Deploy this bundle:
```yaml
bundle: kubernetes
applications:
  loki:
    charm: loki-k8s
    channel: edge
    revision: 99
    series: focal
    resources:
      loki-image: 82
    scale: 1
    constraints: arch=amd64
    storage:
      active-index-directory: kubernetes,1,1024M
      loki-chunks: kubernetes,1,1024M
    trust: true
  pushgateway:
    charm: ../../prometheus-pushgateway-k8s_ubuntu-22.04-amd64.charm
    scale: 1
    constraints: arch=amd64
    resources:
      pushgateway-image: ghcr.io/canonical/prometheus-pushgateway:dev
    storage:
      pushgateway-store: kubernetes,1,1024M
    trust: true
relations:
- - pushgateway:log-proxy
  - loki:logging
```

... and check there are logs in Loki:


```shell
curl -G "http://LOKI_IP:3100/loki/api/v1/query_range" --data-urlencode 'query={juju_charm="prometheus-pushgateway-k8s"} |= ``' 
```
